rollup/presets/package-typescript.js

```diff
@@ -1,16 +1,21 @@
 import babel from '@rollup/plugin-babel';
 import typescript from '@rollup/plugin-typescript';
 import { terser } from 'rollup-plugin-terser';
+import path from 'path';
 import { packageBabelPreset } from '../configs/babel-presets';
 import { externalsForPlugin } from '../configs/externals';
 
-export function packageTypescript() {
-	return [
-		{
-			input: 'src/index.ts',
+export function packageTypescript(pkg) {
+	return Object.keys(pkg.exports).map((exportKey) => {
+		const cjsFileName = pkg.exports[exportKey].require;
+		const mjsFileName = pkg.exports[exportKey].import;
+		const input = path.resolve('/src/', exportKey, './index.ts').slice(1);
```

Comment:
If `src/` is used instead, paths become relative to the running script which is undesirable. That's why it's being sliced after the resolving happens.